src/psyclone/psyir/nodes/omp_directives.py

```diff
@@ -232,6 +232,83 @@ def end_string(self):
         return "omp end single"
 
 
+class OMPMasterDirective(OMPSingleDirective):
+    '''
+    Class representing an OpenMP MASTER directive in the PSyclone AST.
+
+    :param list children: list of Nodes that are children of this Node.
+    :param parent: the Node in the AST that has this directive as a child.
+    :type parent: :py:class:`psyclone.psyir.nodes.Node`
+
+    '''
+    def __init__(self, children=None, parent=None):
+
+        # Call the init method of the base class
+        super(OMPMasterDirective, self).__init__(children=children,
+                                                 parent=parent, nowait=False)
+
+    @property
+    def dag_name(self):
+        '''
+        :returns: the name to use in the DAG for this node.
+        :rtype: str
+        '''
+        _, position = self._find_position(self.ancestor(Routine))
+        return "OMP_master_" + str(position)
+
+    def node_str(self, colour=True):
+        '''
+        Returns the name of this node with (optional) control codes
+        to generate coloured output in a terminal that supports it.
+
+        :param bool colour: whether or not to include colour control codes.
+
+        :returns: description of this node, possibly coloured.
+        :rtype: str
+        '''
+        return self.coloured_name(colour) + "[OMP master]"
+
+    def gen_code(self, parent):
```

Comment:
Parent docstring missing :)

Reply:
I copied this in from the OMPDoDirective - I assume its correct but I'm not sure (I also added this to the OMPSingleDirective).